src/svg/qsvggenerator.cpp

```diff
@@ -934,6 +942,17 @@ bool QSvgPaintEngine::end()
     Q_D(QSvgPaintEngine);
 
     d->stream->setString(&d->defs);
+
+	for (auto it = clip_path_to_id.begin(); it != clip_path_to_id.end(); it++)
+	{
+		std::string path = it->first;
+		int path_id = it->second;
+
+		*d->stream << "<clipPath id=\"clip" << path_id << "\">" << '\n';
+		*d->stream << "\t" << path.c_str() << " \"/> \n";
```

Comment:
Seems like this line is closing the '<path' tag opened it updateState. Why not just close it there?

Actually come to think of it, why not create this entire clipPath element content in that method, stream it to d->defs at that point, then not have to modify this method at all? There's no way in the current implementation the path can change after this point right?